concourse/pipelines/templates/build_pipeline-tpl.yml

```
@@ -1,1842 +1,326 @@
-  - in_parallel:
-    - get: pxf_src
-      passed: [testing_gate_for_pxf-gp]
-      trigger: true
-    - get: pxf_package
-      resource: pxf6_6_gp6_rhel7_released   # for upgrade test
-    - get: pxf_tarball
-      resource: pxf_gp[[gp_ver]]_tarball_rhel7
-      passed: [testing_gate_for_pxf-gp]
-    - get: gpdb_package
-      resource: gpdb[[gp_ver]]_rhel7_rpm_latest-0
-      passed: [testing_gate_for_pxf-gp]
-    - get: gpdb[[gp_ver]]-pxf-dev-centos7-image
-    - get: pxf-automation-dependencies
-    - get: singlecluster
-      resource: singlecluster-hdp2
-  - task: test_pxf-gp[[gp_ver]]-hdp2-new-extension_on_rhel7
-    file: pxf_src/concourse/tasks/upgrade_extension.yml
-    image: gpdb[[gp_ver]]-pxf-dev-centos7-image
-    params:
-      ACCESS_KEY_ID: ((tf-machine-access-key-id))
-      GP_VER: [[gp_ver]]
-      GROUP: pxfExtensionVersion2
-      SECOND_GROUP: pxfExtensionVersion2_1
-      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+[[macros.add_job('jobs/testing-gate-tpl.yml', job_names, environment, build_test_pxf_combinations)]]
+
+[[macros.reset_passed_jobs(build_test_pxf_combinations, job_names)]]
+
+## ---------- Extended Feature (clouds, distros, jdk) tests ----------
+{# include a non-FDW test job for all clouds if supported #}
+{% call(x) macros.for_each_config(build_test_pxf_combinations) %}
+    {% do x.update({ 'passed': '[testing-gate-for-pxf-gp]' }) %}
+    {# iterate over features defined for the given swimlane #}
+    {% for test_feature in x.test_features %}
+        {% do x.update({'use_fdw': 'false'}) %}
+
+        {# if the purpose of the test is to try a different distro/cloud, use it, otherwise use a default distro for the job name #}
+        {% set target = test_feature if (test_feature in distros or test_feature in clouds) else default_distro %}
+        {% do x.update({'target': target}) %}
+
+        {# set the distro to use for the test job's single cluster, use default if the feature are not a distro #}
+        {% set distro = test_feature if test_feature in distros else default_distro %}
+        {% do x.update({'distro': distro}) %}
+
+        {# request a no-impers job for clouds and the default distro (as basic test job is default distro with impersonation) #}
+        {% set use_impers = 'false' if (test_feature in clouds or test_feature == default_distro) else 'true' %}
+        {% do x.update({'use_impers': use_impers}) %}
+
+        {# define a protocol if the feature is a cloud test #}
+        {% set protocol = test_feature if test_feature in clouds else '' %}
+        {% do x.update({'protocol': protocol}) %}
+
+        {# define a feature name to go into the test job name, unless it is a test against a distro/cloud only #}
+        {% set feature = test_feature if test_feature not in distros and test_feature not in clouds else '' %}
+        {% do x.update({'feature': feature}) %}
+        {% include 'jobs/test-tpl.yml' %}
+        [[ macros.add_passing_job(x, job_names, 'extended') ]]
```

```suggestion
        [[macros.add_passing_job(x, job_names, 'extended')]]
```
Also, I recall we wanted to combine including the template with `add_job` macro, did that not work ?